### ISSUE TITLE
Add analytics support

### ISSUE THREAD

**Issue by stevejalim** [mozmeao/birdbox#39]
We're initially assuming we'll go ahead with GA using GTM

In the future (separate work) we'll add a Glean option if we can

**Comment by stevejalim** [comment]
@stephaniehobson The current future.mozilla.org site is running GA via GTM. I'm assuming we can just lift the current GTM/GA markup off the current site and just run it on the Birdbox version. How risky is that assumption, please?

**Comment by stephaniehobson** [comment]
TL;DR: For something out-of-the-box like this I think GTM is an extra unnecessary layer. I think we can rely on gtag instead of GTM.

By default gtag will record:
- page views
- scrolls past 90% of the page length
- clicks on links that lead off the site
- video engagement
- file downloads (as long as they match their list of file extensions)
- form interactions**

** this doesn't work with our newsletter form if you borrowed that code, but I think I know how to fix it

If that list of things is enough we really don't need GTM.By default all GTM does is insert the gtag code! But you have to set up both GA and GTM and then you have to configure GTM with the GA stuff.

GTM is a tool for tracking more complicated interactions (add to shopping cart, things like that). Its promise was that it would be easier than writing code and while it might be easier than writing code I wouldn't call it easy 😝 Also, GA4 now also has the ability to do some simple custom event tracking so GA might be able to handle some additional needs.

If you want to talk about specific use cases, let me know 🙂

Here's the code snippet for gtag.

```
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-XXXXXXX');
</script>
```

**Comment by stephaniehobson** [comment]
... 20 minutes later it occurs to me we may have a language misunderstanding since both the GTM code and the gtag code use the googletagmanager.com URL in their configuration 😅 I think that ^^ might be what futures is doing, in which case, carry on.

**Comment by stevejalim** [comment]
Thanks @stephaniehobson !

Yeah it does look like the current site is using gtag - https://future.mozilla.org/scripts/analytics.0c87ec73.js

> ** this doesn't work with our newsletter form if you borrowed that code, but I think I know how to fix it

Would gtag break the page for the newsletter form (which is indeed lifted from Bedrock) or just not capture what we'd expect it to?

**Comment by stephaniehobson** [comment]
It will just not capture the form submission.

**Comment by stephaniehobson** [comment]
If you [delete this line](https://github.com/mozmeao/birdbox/blob/7157eb12f6a8f39bbaa89b27c076430755bb587a/src/js/newsletter/newsletter.es6#L154) the GA4 form tracking should start working.
